streaming/redis.js

```diff
@@ -4,44 +4,114 @@ import { parseIntFromEnvValue } from './utils.js';
 
 /**
  * @typedef RedisConfiguration
- * @property {import('ioredis').RedisOptions} redisParams
- * @property {string} redisPrefix
+ * @property {string|undefined} redisNamespace
  * @property {string|undefined} redisUrl
+ * @property {import('ioredis').RedisOptions} redisOptions
  */
 
+/**
+ *
+ * @param {NodeJS.ProcessEnv} env
+ * @returns {boolean}
+ */
+function hasSentinelConfiguration(env) {
+  return (
+    typeof env.REDIS_SENTINELS === 'string' &&
+    env.REDIS_SENTINELS.length > 0 &&
+    typeof env.REDIS_SENTINEL_MASTER === 'string' &&
+    env.REDIS_SENTINEL_MASTER.length > 0
+  );
+}
+
+/**
+ *
+ * @param {NodeJS.ProcessEnv} env
+ * @param {import('ioredis').SentinelConnectionOptions} commonOptions
+ * @returns {import('ioredis').SentinelConnectionOptions}
+ */
+function getSentinelConfiguration(env, commonOptions) {
+  const redisDatabase = parseIntFromEnvValue(env.REDIS_DB, 0, 'REDIS_DB');
+  const sentinelPort = parseIntFromEnvValue(env.REDIS_SENTINEL_PORT, 26379, 'REDIS_SENTINEL_PORT');
+
+  const sentinels = env.REDIS_SENTINELS.split(',').map((sentinel) => {
+    const [host, port] = sentinel.split(':', 2);
+
+    /** @type {import('ioredis').SentinelAddress} */
+    return {
+      host: host,
+      port: port ?? sentinelPort,
+      // Force support for both IPv6 and IPv4, by default ioredis sets this to 4,
+      // only allowing IPv4 connections:
+      // https://github.com/redis/ioredis/issues/1576
+      family: 0
+    };
+  });
+
+  return {
+    db: redisDatabase,
+    name: env.REDIS_SENTINEL_MASTER,
+    username: env.REDIS_USERNAME,
+    password: env.REDIS_PASSWORD,
+    sentinelUsername: env.REDIS_SENTINEL_USERNAME ?? env.REDIS_USERNAME,
+    sentinelPassword: env.REDIS_SENTINEL_PASSWORD ?? env.REDIS_PASSWORD,
+    sentinels,
+    ...commonOptions,
+  };
+}
+
 /**
  * @param {NodeJS.ProcessEnv} env the `process.env` value to read configuration from
  * @returns {RedisConfiguration} configuration for the Redis connection
  */
 export function configFromEnv(env) {
-  // ioredis *can* transparently add prefixes for us, but it doesn't *in some cases*,
-  // which means we can't use it. But this is something that should be looked into.
-  const redisPrefix = env.REDIS_NAMESPACE ? `${env.REDIS_NAMESPACE}:` : '';
-
-  let redisPort = parseIntFromEnvValue(env.REDIS_PORT, 6379, 'REDIS_PORT');
-  let redisDatabase = parseIntFromEnvValue(env.REDIS_DB, 0, 'REDIS_DB');
+  const redisNamespace = env.REDIS_NAMESPACE ? `${env.REDIS_NAMESPACE}:` : '';
```

Comment:
This does actually need to be an empty string and not `undefined` due to how we're constructing channel names for `subscribe`/`unsubscribe` commands